### DISCUSSION TITLE
integer value of json format in request body will change to string in v0.110 but not v109

### DISCUSSION THREAD

**Discussion by Redias** [fastapi/fastapi#11292] [Questions]
type annotation: 
    async def list(request_body: Dict[str, str|int])

python 3.10.11
fastapi v0.109/v0.110
uvicorn v27.0/v27.1

test tool: postman v10.24
           application/json

**Comment by YuriiMotov** [comment]
Can you give a code example?
I tried this code and it passes tests on both 0.109.0 and 0.110.0:
```
from fastapi import FastAPI
from fastapi.testclient import TestClient

app: FastAPI = FastAPI()

@app.post("/int_str")
async def int_str(v: dict[str, str | int]):
    return {"type": type(v["int_or_str"]).__name__}

client = TestClient(app)

def test_int():
    resp = client.post("/int_str", json={"int_or_str": 1})
    assert resp.status_code == 200, resp.text
    assert resp.json()["type"] == "int"

def test_str():
    resp = client.post("/int_str", json={"int_or_str": "1"})
    assert resp.status_code == 200, resp.text
    assert resp.json()["type"] == "str"
```

There is a similar issue, but it's not related to FastAPI version, it's related to Pydantic version: https://github.com/tiangolo/fastapi/issues/11251

**Reply by Redias** [reply]
Thanks for replying, I test your code in my new virtual environment and the results are right. But in the production environment, the error  still exists. I 'll test `poetry` and local-pypi-mirror etc... If there are some new issues, I 'll feedback. Thank you !
